File: CS-Examples/08_FilteringAndValidation/GetSettingsOfDataValidation.cs
using System;
using System.Data.OleDb;
using System.Drawing;
using System.Collections;
using System.ComponentModel;
using System.Windows.Forms;
using System.Data;
using System.IO;
using Spire.Xls;
using Spire.Xls.Charts;
using System.Text;
using System.Collections.Generic;

namespace GetSettingsOfDataValidation
{
	public partial class Form1 : Form
	{
        public Form1()
        {
            InitializeComponent();
        }
		private void btnRun_Click(object sender, System.EventArgs e)
        {
            //Create a workbook
            Workbook workbook = new Workbook();

            //Load the Excel document from disk
            workbook.LoadFromFile(@"..\..\..\..\..\..\Data\Sample.xlsx");

            //Get first worksheet of the workbook
            Worksheet worksheet = workbook.Worksheets[0];

            //Cell B4 has the Decimal Validation
            CellRange cell = worksheet.Range["B4"];

            //Get the valditation of this cell
            Validation validation = cell.DataValidation;

            //Get the settings
            string allowType = validation.AllowType.ToString();
            string data = validation.CompareOperator.ToString();
            string minimum = validation.Formula1.ToString();
            string maximum = validation.Formula2.ToString();
            string ignoreBlank = validation.IgnoreBlank.ToString();
           
            //Create StringBuilder to save 
            StringBuilder content = new StringBuilder();

            //Set string format for displaying
            string result = string.Format("Settings of Validation: \r\nAllow Type: " + allowType + "\r\nData: " + data + "\r\nMinimum: " + minimum +"\r\nMaximum: " + maximum + "\r\nIgnoreBlank: "+ignoreBlank);

            //Add result string to StringBuilder
            content.AppendLine(result);

            //String for output file 
            String outputFile = "Output.txt";

            //Save them to a txt file
            File.WriteAllText(outputFile, content.ToString());

            //Launching the output file.
            Viewer(outputFile);
		}
		private void Viewer( string fileName )
		{
			try
			{
				System.Diagnostics.Process.Start(fileName);
			}
			catch{}
		}
        private void btnClose_Click(object sender, EventArgs e)
        {
            Close();
        }

	}
}

File: VB-Examples/08_FilteringAndValidation/GetSettingsOfDataValidation.vb
Imports System.Data.OleDb
Imports System.Collections
Imports System.ComponentModel
Imports System.IO
Imports Spire.Xls
Imports Spire.Xls.Charts
Imports System.Text

Namespace GetSettingsOfDataValidation
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub
		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
            ' Create a new Workbook object
            Dim workbook As New Workbook()

            ' Load an existing Excel file named "Sample.xlsx" from the specified path
            workbook.LoadFromFile("..\..\..\..\..\..\Data\Sample.xlsx")

            ' Get the first worksheet (index 0) from the workbook
            Dim worksheet As Worksheet = workbook.Worksheets(0)

            ' Specify the cell B4 for which we want to retrieve the validation settings
            Dim cell As CellRange = worksheet.Range("B4")

            ' Get the data validation settings for the specified cell
            Dim validation As Validation = cell.DataValidation

            ' Retrieve various settings from the validation object
            Dim allowType As String = validation.AllowType.ToString()
            Dim compareOperator As String = validation.CompareOperator.ToString()
            Dim formula1 As String = validation.Formula1.ToString()
            Dim formula2 As String = validation.Formula2.ToString()
            Dim ignoreBlank As String = validation.IgnoreBlank.ToString()

            ' Create a StringBuilder object to save the output content
            Dim content As New StringBuilder()

            ' Set the string format for displaying the validation settings
            Dim result As String = String.Format("Settings of Validation: " & vbCrLf & "Allow Type: " & allowType & vbCrLf & "Compare Operator: " & compareOperator & vbCrLf & "Formula 1: " & formula1 & vbCrLf & "Formula 2: " & formula2 & vbCrLf & "Ignore Blank: " & ignoreBlank)

            ' Add the result string to the StringBuilder
            content.AppendLine(result)

            ' Specify the output file name as "Output.txt"
            Dim outputFile As String = "Output.txt"

            ' Save the contents of the StringBuilder to a text file
            File.WriteAllText(outputFile, content.ToString())
            ' Release the resources used by the workbook
            workbook.Dispose()

            'Launching the output file.
            Viewer(outputFile)
		End Sub
		Private Sub Viewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub
		Private Sub btnClose_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnClose.Click
			Close()
		End Sub

	End Class
End Namespace
